### PR TITLE
Added smoothing and confusion matrix and more

### PR DESCRIPTION
**Pull request by HansBambel**
This pull request adds:

- [x] Savitzky-Golay smoothing (on by default)
- [x] A confusion matrix for later plotting
- [x] Add precision, recall and F1 score
- [x] Inspect dataset now has a nicer output and shows class-proportions
- [x] added learning-rate scheduler that decays learning rate after X epochs of no improvement in val_loss (default is 5 epochs)
- [x] added evaluation on test-set (normalization is done with training set that is randomly split into train-val. Therefore small differences may occur)
- [x] **important Fix: used train dataset as validation set**

Requirements:
Removed numba, but added scipy (due to smoothing)

### CODE REVIEW

**Comment by HansBambel** [comment]
Do not merge yet. I will implement precision, recall and F1 score
